components/script/dom/htmlstyleelement.rs

```diff
@@ -229,11 +246,26 @@ impl VirtualMethods for HTMLStyleElement {
         }
 
         if context.tree_connected {
-            if let Some(s) = self.stylesheet.borrow_mut().take() {
-                self.clean_stylesheet_ownership();
-                self.stylesheet_list_owner()
-                    .remove_stylesheet(self.upcast(), &s)
+            self.remove_stylesheet();
+        }
+    }
+
+    fn attribute_mutated(&self, attr: &Attr, mutation: AttributeMutation) {
+        if let Some(s) = self.super_type() {
+            s.attribute_mutated(attr, mutation);
+        }
+
+        if attr.name() == "type" &&
+            self.upcast::<Node>().is_in_a_document_tree() &&
+            !self.in_stack_of_open_elements.get()
+        {
+            if let AttributeMutation::Set(Some(old_value)) = mutation {
+                if **old_value == **attr.value() {
+                    return;
+                }
             }
+            self.remove_stylesheet();
+            self.parse_own_css();
```

Comment:
Where is this part specified? The `update a style block` algorithm doesn't run again when the type attribute is modified, right?